src/pip/_internal/commands/freeze.py

```diff
@@ -27,46 +28,54 @@ def __init__(self, *args, **kw):
         super(FreezeCommand, self).__init__(*args, **kw)
 
         self.cmd_opts.add_option(
-            '-r', '--requirement',
+            '-r',
+            '--requirement',
             dest='requirements',
             action='append',
             default=[],
             metavar='file',
             help="Use the order in the given requirements file and its "
-                 "comments when generating output. This option can be "
-                 "used multiple times.")
+            "comments when generating output. This option can be "
+            "used multiple times.",
+        )
         self.cmd_opts.add_option(
-            '-f', '--find-links',
+            '-f',
+            '--find-links',
             dest='find_links',
             action='append',
             default=[],
             metavar='URL',
-            help='URL for finding packages, which will be added to the '
-                 'output.')
+            help='URL for finding packages, which will be added to the ' 'output.',
```

Comment:
This is awful. If black is going to put adjacent string literals onto a single line, it needs to merge them.

Reply:
https://github.com/ambv/black/issues/26

Reply:
Yea, that issue is basically asking for this, and the summary of that is basically "it has edge cases like strings of different prefixes" but I think the bigger issue is that a strong invariant that black keeps right now is that the AST before black modifies the code matches the AST after black modifies the code. This is basically the way that black ensures that it's changes are purely format related and it is not changing the actual meaning of the code. In Python merging these strings changes the AST, so that's why black doesn't do it.

In practice, this sort of thing doesn't happen organically that often, typically your editor knows the line length and you just break as appropriate. It tends to happen when you're changing the length of the line because that changes what fits onto one line, so somethings that required a break before, no longer do.